src/Trace/TraceSpan.php

```diff
@@ -0,0 +1,256 @@
+     */
+    private $startTime;
+
+    /**
+     * End time of the span in nanoseconds in "Zulu" format.
+     * @var string
+     */
+    private $endTime;
+
+    /**
+     * ID of the parent span if any.
+     * @var string
+     */
+    private $parentSpanId;
+
+    /**
+     * Collection of labels attached to this span.
+     * @var array
+     */
+    private $labels;
+
+    /**
+     * Instantiate a new TraceSpan instance.
+     *
+     * @param array $options [optional] {
+     *      Configuration options.
+     *
+     *      @type string $spanId The ID of the span. If not provided,
+     *            one will be generated automatically for you.
+     *      @type string $kind Distinguishes between spans generated
+     *            in a particular context. **Defaults to**
+     *            SPAN_KIND_UNSPECIFIED.
+     *      @type string $name The name of the span.
+     *      @type string $startTime Start time of the span in
+     *            nanoseconds in "Zulu" format.
+     *      @type string $endTime End time of the span in
+     *            nanoseconds in "Zulu" format.
+     *      @type string $parentSpanId ID of the parent span if any.
+     *      @type array $labels Associative array of $label => $value
+     *            to attach to this span.
+     * }
+     */
+    public function __construct($options = [])
+    {
+        if (array_key_exists('spanId', $options)) {
+            $this->spanId = $options['spanId'];
+        } else {
+            $this->spanId = $this->generateSpanId();
+        }
+
+        if (array_key_exists('kind', $options)) {
+            $this->kind = $options['kind'];
+        } else {
+            $this->kind = self::SPAN_KIND_UNSPECIFIED;
+        }
+
+        if (array_key_exists('name', $options)) {
+            $this->name = $options['name'];
+        } else {
+            $this->name = $this->generateSpanName();
+        }
+
+        if (array_key_exists('startTime', $options)) {
+            $this->startTime = $options['startTime'];
+        }
+
+        if (array_key_exists('endTime', $options)) {
+            $this->startTime = $options['endTime'];
+        }
+
+        if (array_key_exists('parentSpanId', $options)) {
+            $this->parentSpanId = $options['parentSpanId'];
+        }
+
+        if (array_key_exists('labels', $options)) {
+            $this->labels = $options['labels'];
+        } else {
+            $this->labels = [];
+        }
+    }
+
+    /**
+     * Set the start time for this span.
+     *
+     * @param  \DateTime $when [optional] The start time of this span.
+     *         **Defaults to** now.
+     */
+    public function start(\DateTime $when = null)
+    {
+        $this->startTime = $this->formatDate($when);
+    }
+
+    /**
+     * Set the end time for this span.
+     *
+     * @param  \DateTime $when [optional] The end time of this span.
+     *         **Defaults to** now.
+     */
+    public function finish(\DateTime $when = null)
```

Comment:
Your naming for getters and setters should be inverted.

`finish()` = getter
`setFinish()` = setter.